app/models/post.rb

```diff
@@ -0,0 +1,21 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
+  has_many :likes, foreign_key: 'post_id'
+  has_many :comments, foreign_key: 'post_id'
+
+  after_save :update_post_counter
+
+  private
+
+  def update_post_counter
+    author.increment!(:posts_counter)
+  end
+
+  def post_counter
+    author.update(posts_counter: author.posts.count)
+  end
+
+  def return_most_comment
+    comments.order(created_at: :desc).limit(5)
+  end
+end
```

Comment:
- [ ] Well done for using [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) 🥇🥇 as suggested by the reviewer as OPTIONAL However, because of its implementation, now I am unable to get the 5 recent comments  😓 and that's because you are using it as a private method ⚠

![image](https://github.com/vanina123/blog-app/assets/106119105/6dd132ef-78b6-42a2-8c76-401db55b748a)

During the previous review, it was like this 👇
![image](https://github.com/vanina123/blog-app/assets/106119105/9e8ab4f2-a891-480d-9f3e-f1fc607d0248)


Your can consider using something like below 👇

```
  after_save :update_post_counter
  
  def return_most_comment
    comments.order(created_at: :desc).limit(5)
  end

  private

  def update_post_counter
    author.increment!(:posts_counter)
  end
```